docs/Makefile

```diff
@@ -18,6 +18,7 @@ all: dirhtml
 
 clean:
 	rm -rf '$(BUILDDIR)'/*
+	rm -f poetry.lock
```

Comment:
I don't think this belongs here. `clean` is meant to clean build files which can always be regenerated to the same output, `poetry.log` actually holds information (snapshot of all the moving references in a certain time), same as go.sum.

Reply:
I followed the [upgrade guide](https://sphinx-theme.scylladb.com/stable/upgrade/1-3-to-1-4.html), but personally I agree, it sort of defeats the purpose of even having the `.lock` file. @annastuchlik why is that considered a "breaking change" in the upgrade guide?

Reply:
> I followed the upgrade guide
We don't follow their Makefile, nor CI for differences on opinions on what reproducibility means or make should look like. Totally fine to differ in `clean` target.

Having to remove `.lock` probably suggest something wrong is being done somewhere else but even so it should be one shot, not a part of regular cleanup.

Reply:
I agree. I removed the Makefile change.